toonz/sources/tnztools/filltool.cpp

```diff
@@ -252,6 +252,99 @@ class VectorRectFillUndo final : public TToolUndo {
   int getHistoryType() override { return HistoryType::FillTool; }
 };
 
+//=============================================================================
+// VectorGapSizeChangeUndo
+//-----------------------------------------------------------------------------
+
+class VectorGapSizeChangeUndo final : public TToolUndo {
+  double m_oldGapSize;
+  double m_newGapSize;
+  int m_row;
+  int m_column;
+  TVectorImageP m_vi;
+  std::vector<TFilledRegionInf> m_oldFillInformation;
+
+public:
+  VectorGapSizeChangeUndo(double oldGapSize, double newGapSize,
+                          TXshSimpleLevel *sl, const TFrameId &fid,
+                          TVectorImageP vi,
+                          std::vector<TFilledRegionInf> oldFillInformation)
+      : TToolUndo(sl, fid)
+      , m_oldGapSize(oldGapSize)
+      , m_newGapSize(newGapSize)
+      , m_oldFillInformation(oldFillInformation)
+      , m_vi(vi) {
+    TTool::Application *app = TTool::getApplication();
+    if (app) {
+      m_row    = app->getCurrentFrame()->getFrame();
+      m_column = app->getCurrentColumn()->getColumnIndex();
+    }
+    ImageUtils::getFillingInformationInArea(vi, m_oldFillInformation,
+                                            vi->getBBox());
```

Comment:
I'm not sure why you update `m_oldFillInformation` here.
It seems that `vi` is already modified with new gap size and calling `getFillingInformationInArea()` overwrites the old fill information with the new one.

Reply:
Thanks for catching this.  It was from a previous attempt.